### PR TITLE
Add unit tests for the util package

### LINKED ISSUE #297: Support Azure DevOps Git Repo
Git repos hosted in VSTS now have new urls in Azure DevOps.

VSTS: abc.visualstudio.com
Azure DevOps: dev.azure.com/abc

Need to update git context parser to handle the new url format.

### PR DESCRIPTION
**Pull request by ehotinger**
**Purpose of the PR:**

- This PR brings the test coverage of the `util` package from 70.8% to 100%.
- Adds support for Azure DevOps URLs.
- Run `go fmt` on `setup_windows.go`

**Fixes #297**

### CODE REVIEW

File: util/url_test.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ func TestIsVstsGitURL(t *testing.T) {
 		url      string
 		expected bool
 	}{
+		{"   https://azure.visualstudio.com/ACR/_git/Build/", false}, // leading spaces cause a parse error
```
Comment (sajayantony) [inline util/url_test.go]: Can you include the pattern for azure pipelines as well?

Comment (ehotinger) [reply util/url_test.go]: As a part of https://github.com/Azure/acr-builder/issues/297 we can do that.

Comment (ehotinger) [reply util/url_test.go]: Updated and put that in this PR as well.

**Comment by sajayantony** [comment]
Might be worth adding a unicode domain in the test as well.

File: util/url.go [hunk 2]
```diff
@@ -10,6 +10,18 @@ import (
...
 )
 
+// IsAzureDevOpsGitURL determines whether or not the specified string is an Azure DevOps Git URL.
+func IsAzureDevOpsGitURL(s string) bool {
+	url, err := url.Parse(strings.ToLower(s))
+	if err != nil {
+		return false
+	}
+	return url.Scheme == "https" &&
+		strings.Contains(url.Host, "dev.azure.com") &&
```
Comment (northtyphoon) [inline util/url.go]: nit: check equals

**Comment by northtyphoon** [comment]
It just occurs to me that we may need to port the AzureDevOps Git Repo fix to stable branch for windows. An alternative solution is to switch windows build to mater branch if the build part is ready. We can sync on Team.

**Pull request merged**